presto-tpch/src/main/java/com/facebook/presto/tpch/StatisticsEstimator.java

```diff
@@ -0,0 +1,258 @@
+    {
+        return bigStatistics.getColumns().entrySet().stream().collect(toImmutableMap(
+                Map.Entry::getKey,
+                entry -> {
+                    String columnName = entry.getKey();
+                    TpchColumn<?> column = table.getColumn(columnName);
+                    ColumnStatisticsData bigColumnStatistics = entry.getValue();
+                    ColumnStatisticsData smallColumnStatistics = smallStatistics.getColumns().get(columnName);
+                    return rescale(column, bigColumnStatistics, smallColumnStatistics, scaleFactor);
+                }
+        ));
+    }
+
+    private ColumnStatisticsData rescale(TpchColumn<?> column, ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (UNSUPPORTED_COLUMNS.contains(column)) {
+            return ColumnStatisticsData.empty();
+        }
+        else {
+            return rescale(big, small, scaleFactor);
+        }
+    }
+
+    private ColumnStatisticsData rescale(ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (areClose(big, small)) {
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount(),
+                    big.getNullsCount(),
+                    checkPresent(withBoth(big.getLowestValue(), small.getLowestValue(), this::checkClose)),
+                    checkPresent(withBoth(big.getHighestValue(), small.getHighestValue(), this::checkClose))
+            );
+        }
+        else {
+            Function<Number, Double> scale = value -> value.doubleValue() * scaleFactor;
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount().map(scale.andThen(Number::longValue)),
+                    big.getNullsCount().map(scale.andThen(Number::longValue)),
+                    Types.tryCast(big.getLowestValue(), Number.class),
+                    Types.tryCast(big.getHighestValue(), Number.class).map(scale));
+        }
+    }
+
+    private boolean areClose(ColumnStatisticsData big, ColumnStatisticsData small)
+    {
+        Optional<Long> smallCount = small.getUniqueValuesCount();
+        Optional<Long> bigCount = big.getUniqueValuesCount();
+
+        return smallCount.isPresent() == bigCount.isPresent() &&
+                withBoth(smallCount, bigCount, this::areClose)
+                        .orElse(boundsAreClose(big, small));
+    }
+
+    private Boolean boundsAreClose(ColumnStatisticsData big, ColumnStatisticsData small)
+    {
+        return withBoth(small.getLowestValue(), big.getLowestValue(), this::areClose)
+                .flatMap(result -> withBoth(small.getHighestValue(), big.getHighestValue(), this::areClose))
+                .orElse(false);
+    }
+
+    private Object checkClose(Object leftValue, Object rightValue)
```

Comment:
checkNotNull does, though. `requireNonNull` is the only require-method in Objects, so I'd say it's not the matter  of java/guava or require/check, but of 'can i return a value because I get it as arg or not'